src/api/mod.rs

```diff
@@ -10,16 +11,16 @@ use plume_models::{
 };
 
 #[derive(FromForm)]
-struct OAuthRequest {
+pub struct OAuthRequest {
     client_id: String,
     client_secret: String,
     password: String,
     username: String,
     scopes: String,
 }
 
-#[get("/oauth2?<query>")]
-fn oauth(query: OAuthRequest, conn: DbConn) -> Json<serde_json::Value> {
+#[get("/oauth2?<query..>")]
```

Comment:
what is the `..` in a query part for? I only know its use in the path part

Reply:
It is the new way to handle queries in Rocket 0.4. The new syntax is:

- `/?<x>` is a single query argument (`?x=foo`)
- You can separate query arguments with `&` (`/?<x>&<y>` will match `?x=foo&y=bar`)
- If you want to deserialize a whole struct, you have to use `..` and each of its field will be a query argument (I think it was the default behavior in 0.3)